indico_toolkit/indico_wrapper/pred_manipulation.py

```diff
@@ -0,0 +1,115 @@
+from typing import List
+
+
+class ManipulatePreds:
+    """
+    Class to help expand prediction text and indexing via OCR data.
+    """
+
+    def __init__(self, ocr_tokens: List[dict], preds: List[dict]):
+        self.ocr_tokens = ocr_tokens
+        self.predictions = preds
+
+    def expand_predictions(self, pred_start: int, pred_end: int) -> dict:
+        """
+        Expand predictions and boundaries to match that of OCR data.
+
+        Args:
+            pred_start (int): Starting prediction index
+            pred_end (int): Ending prediction index
+        Returns:
+            dict: Returns expanded prediction dictionary
+        """
+        expanded_start, expanded_end = pred_start, pred_end
+        pred_index = None
+
+        # Find index value of bounded prediction
+        for index, pred in enumerate(self.predictions):
+            if pred["start"] == pred_start or pred["end"] == pred_end:
+                pred_index = index
+                break
+
+        if pred_index is None:
+            raise ValueError("No matching prediction found.")
+
+        # Validate current prediction needs expanding
+        original_text = self.predictions[pred_index]["text"]
+        ocr_text_initial = self._get_ocr_text(pred_start, pred_end)
+        if original_text == ocr_text_initial:
+            print("No expansion needed")
+            return self.predictions[pred_index]
+
+        # Use overlapping boundaries and expand text / boundaries to match OCR data
+        for token in self.ocr_tokens:
+            if token["doc_offset"]["start"] <= pred_start <= token["doc_offset"]["end"]:
+                expanded_start = min(expanded_start, token["doc_offset"]["start"])
+            if token["doc_offset"]["start"] <= pred_end <= token["doc_offset"]["end"]:
+                expanded_end = max(expanded_end, token["doc_offset"]["end"])
+
+        expanded_text = self._get_ocr_text(expanded_start, expanded_end)
+        if expanded_text != ocr_text_initial:
+            raise ValueError("Expanded text does not match the OCR text.")
+
+        if expanded_text == ocr_text_initial:
+            # Update prediction
+            self.predictions[pred_index]["start"] = expanded_start
+            self.predictions[pred_index]["end"] = expanded_end
+            self.predictions[pred_index]["text"] = expanded_text
+
+        return self.predictions[pred_index]
+
+    def is_token_nearby(
+        self, ocr_start: int, ocr_end: int, search_tokens: List[str], distance: int
```

Comment:
I'm not sure I understand what distance means here or when this function would be useful (also don't like the idea of entering in the ocr_start ocr_end (how would you know what those values should be?)

Reply:
I think the more relevant functionality would be "is SOME_TEXT contained within a token within X distance"